src/FisherKolmogorov.hpp

```diff
@@ -0,0 +1,252 @@
+#ifndef HEAT_NON_LINEAR_HPP
+#define HEAT_NON_LINEAR_HPP
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include "DiffusionTensor.hpp"
+
+#include <fstream>
+#include <iostream>
+
+using namespace dealii;
+
+// Class representing the non-linear diffusion problem.
+class FisherKolmogorov
+{
+public:
+  // Physical dimension (1D, 2D, 3D)
+  static constexpr unsigned int dim = 3;
+
+  // Function for the forcing term.
+  class ForcingTerm : public Function<dim>
+  {
+  public:
+    virtual double
+    value(const Point<dim> & p,
+          const unsigned int /*component*/ = 0) const override
+    {
+      if(CONVERGENCE_TEST) {
```

Comment:
I would refrain from using define macros (except in very few occasions), since they are very error prone, for a number of reasons.

This flag could have been passed as a constructor argument, or you could have had two different function classes.